src/components/dialogs/parameters/sensi/sensi-hvdcs.js

```diff
@@ -0,0 +1,169 @@
+    SENSITIVITY_TYPES,
+} from './sensi-parameters-selector';
+import {
+    CONTINGENCIES,
+    HVDC_LINES,
+    ID,
+    NAME,
+    PARAMETER_SENSI_HVDC,
+    SELECTED,
+    SENSITIVITY_TYPE,
+    MONITORED_BRANCHES,
+    FILTER_ID,
+    FILTER_NAME,
+} from '../../../utils/field-constants';
+import yup from '../../../utils/yup-config';
+import { useIntl } from 'react-intl';
+import { useFieldArray } from 'react-hook-form';
+import DndTable from '../../../utils/dnd-table/dnd-table';
+
+export const getSensiHVDCsFormSchema = () => ({
+    [PARAMETER_SENSI_HVDC]: yup.array().of(
+        yup.object().shape({
+            [MONITORED_BRANCHES]: yup.array().of(
+                yup.object().shape({
+                    [ID]: yup.string().required(),
+                    [NAME]: yup.string().required(),
+                })
+            ),
+            [SENSITIVITY_TYPE]: yup.string().nullable(),
+            [HVDC_LINES]: yup.array().of(
+                yup.object().shape({
+                    [ID]: yup.string().required(),
+                    [NAME]: yup.string().required(),
+                })
+            ),
+            [CONTINGENCIES]: yup.array().of(
+                yup.object().shape({
+                    [ID]: yup.string().required(),
+                    [NAME]: yup.string().required(),
+                })
+            ),
+        })
+    ),
+});
+
+export const getSensiHvdcformatNewParams = (newParams) => {
+    return {
+        [PARAMETER_SENSI_HVDC]: newParams.sensitivityHVDC.map(
+            (sensitivityInjection) => {
+                return {
+                    [MONITORED_BRANCHES]: sensitivityInjection[
+                        MONITORED_BRANCHES
+                    ].map((filter) => {
+                        return {
+                            [FILTER_ID]: filter[ID],
+                            [FILTER_NAME]: filter[NAME],
+                        };
+                    }),
+                    [HVDC_LINES]: sensitivityInjection[HVDC_LINES].map(
+                        (filter) => {
+                            return {
+                                [FILTER_ID]: filter[ID],
+                                [FILTER_NAME]: filter[NAME],
+                            };
+                        }
+                    ),
+                    [SENSITIVITY_TYPE]: sensitivityInjection[SENSITIVITY_TYPE],
+                    [CONTINGENCIES]: sensitivityInjection[CONTINGENCIES].map(
+                        (filter) => {
+                            return {
+                                [FILTER_ID]: filter[ID],
+                                [FILTER_NAME]: filter[NAME],
+                            };
+                        }
+                    ),
+                };
+            }
+        ),
+    };
+};
+const SensiHVDCs = () => {
+    const intl = useIntl();
+
+    const useFieldArrayOutput = useFieldArray({
+        name: `${PARAMETER_SENSI_HVDC}`,
+    });
+
+    const COLUMNS_DEFINITIONS = useMemo(() => {
```

Comment:
create a new typescript file to define this

Reply:
done